src/tabs/presets/FavoritePresets.js

```diff
@@ -0,0 +1,118 @@
+
+const s_maxFavoritePresetsCount = 50;
+const s_favoritePresetsListConfigStorageName = "FavoritePresetsList";
+
+
+class FavoritePreset {
+    constructor(presetPath){
+        this.presetPath = presetPath;
+        this.lastPickDate = Date.now();
+    }
+}
+
+
+class FavoritePresetsData {
+    constructor() {
+        this._favoritePresetsList = [];
+    }
+
+    _sort() {
+        this._favoritePresetsList.sort((a, b) => (a.lastPickDate > b.lastPickDate) ? 1 : -1);
+    }
+
+    _purgeOldPresets() {
+        this._favoritePresetsList.splice(s_maxFavoritePresetsCount, this._favoritePresetsList.length);
```

Comment:
Why not `this._favoritePresetsList = [];`

Reply:
this does not clean the list, it just makes sure it does not have more elements than s_maxFavoritePresetsCount